contrib/devtools/lint-tests.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Check the test suite naming convention
+
+NAMING_INCONSISTENCIES=$(git grep -E '^BOOST_FIXTURE_TEST_SUITE\(' -- \
+    "src/test/**.cpp" "src/wallet/test/**.cpp" | \
+    grep -vE '/(.*?)\.cpp:BOOST_FIXTURE_TEST_SUITE\(\1, .*\)$')
+if [[ ${NAMING_INCONSISTENCIES} != "" ]]; then
```

Comment:
There is a serious shell quoting problem here. This isn't just a nit, it changes the semantics of the test. Use test -n and quote the variable, e.g.

```bash
if [ -n "${NAMING_INCONSISTENCIES}" ]; then
    ...
```

Reply:
Actually I have to take this back, apparently [[ doesn't perform word-splitting. Still scares me though.

Reply:
@eklitzke `shellcheck` seems happy :-)

Reply:
No need to be scared :-)  `[[ ${FOO} != "" ]]` is perfectly safe AFAIK. `shellcheck` agrees :-)

Reply:
This is a `bash` script and is explicit about it (`#!/bin/bash`), so no need for follow-up PR:s for other shells :-)